File: LeetCode/Top Interview Questions/1. Two Sum/twoSum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    const numIndexMap = new Map();
    for (let i = 0; i < nums.length; i++) {
        const complement = target - nums[i];
        if (numIndexMap.has(complement)) {
            return [numIndexMap.get(complement), i];
        }
        numIndexMap.set(nums[i], i);
    }
    return [];
};

File: LeetCode/Top Interview Questions/1. Two Sum/twoSum.c
int* twoSum(int* nums, int numsSize, int target, int* returnSize) {
    int* numIndexMap = (int*)malloc(sizeof(int) * 10001);
    for (int i = 0; i < 10001; ++i) {
        numIndexMap[i] = -1;
    }
    for (int i = 0; i < numsSize; ++i) {
        int complement = target - nums[i];
        if (numIndexMap[complement] != -1) {
            int* result = (int*)malloc(sizeof(int) * 2);
            result[0] = numIndexMap[complement];
            result[1] = i;
            *returnSize = 2;
            free(numIndexMap);
            return result;
        }
        numIndexMap[nums[i]] = i;
    }
    *returnSize = 0;
    free(numIndexMap);
    return NULL;
}